dev-proxy-plugins/RequestLogs/OpenApiSpecGeneratorPlugin.cs

```diff
@@ -355,7 +357,15 @@ request.Context is null ||
         {
             var server = openApiDoc.Servers.First();
             var fileName = GetFileNameFromServerUrl(server.Url);
-            var docString = openApiDoc.SerializeAsJson(OpenApiSpecVersion.OpenApi3_0);
+
+            var openApiSpecVersion = _configuration.SpecVersion switch
+            {
+                2 => OpenApiSpecVersion.OpenApi2_0,
+                3 => OpenApiSpecVersion.OpenApi3_0,
+                _ => throw new NotSupportedException($"OpenAPI spec version {_configuration.SpecVersion} is not supported. Supported versions are 2 and 3.")

```

Comment:
oh wait, I expected to see the error on the launch, while we'd throw it while generating the doc. So the error is likely there, but I suggest that we throw it earlier, when loading the plugin, so that we fail early and users won't lose work in the generation stage.